geo-test-fixtures/Cargo.toml

```diff
@@ -5,5 +5,5 @@ edition = "2021"
 publish = false
 
 [dependencies]
-wkt = { version = "0.10.0", default-features = false }
+wkt = { git = "https://github.com/nyurik/wkt.git", branch = "wkt-dim", optional = true }
```

Comment:
Why do you need this?

Reply:
@lnicola try to compile without it. This PR requires the changes from https://github.com/georust/wkt/pull/85

Reply:
I get an error on your branch:

```
error: failed to select a version for `cc`.
    ... required by package `cmake v0.1.48`
    ... which satisfies dependency `cmake = "^0.1.48"` of package `proj-sys v0.22.0 (https://github.com/georust/proj?rev=e39a671d9afd52c13e180516d6dffd9b79a89d2e#e39a671d)`
    ... which satisfies git dependency `proj-sys` of package `proj v0.25.2 (https://github.com/georust/proj?rev=e39a671d9afd52c13e180516d6dffd9b79a89d2e#e39a671d)`
    ... which satisfies git dependency `proj` of package `geo v0.19.0 (/home/grayshade/Projects/georust/geo/geo)`
versions that meet the requirements `^1.0.72` are: 1.0.73, 1.0.72

all possible versions conflict with previously selected packages.

  previously selected package `cc v1.0.71`
    ... which satisfies dependency `cc = "^1.0.62"` (locked to 1.0.71) of package `ring v0.16.20`
    ... which satisfies dependency `ring = "^0.16.20"` (locked to 0.16.20) of package `rustls v0.20.2`
    ... which satisfies dependency `rustls = "^0.20.1"` (locked to 0.20.2) of package `hyper-rustls v0.23.0`
    ... which satisfies dependency `hyper-rustls = "^0.23"` (locked to 0.23.0) of package `reqwest v0.11.9`
    ... which satisfies dependency `reqwest = "^0.11.9"` of package `proj v0.25.2 (https://github.com/georust/proj?rev=e39a671d9afd52c13e180516d6dffd9b79a89d2e#e39a671d)`
    ... which satisfies git dependency `proj` of package `geo v0.19.0 (/home/grayshade/Projects/georust/geo/geo)`
```

Reply:
Your code doesn't build, but I can revert your `wkt` version change, bump some versions (because these changes are actually breaking) and it builds fine for me:

![image](https://user-images.githubusercontent.com/308347/159480834-fea9c238-3740-4182-a16c-f0790e36d10f.png)

So I'm not convinced there's any valid reason to move `wkt` from one repo to another.

Reply:
@lnicola once again, this is not about being able to run `cargo check` locally - it might pass because it is a test compile that fails, not a production one.  This is about being unable to get CI to pass.  Our CI will not allow your code in, so we cannot use `bors` to check it in without some hacking around the issues

Reply:
Why wouldn't the CI allow my changes?

Reply:
I think I am beginning to see the source of the issue -- geo`s CI uses `working-directory` to run tests against just some of the geo's sub-dirs, not to run tests on all workspaces. Except for the `lint` CI target, which runs `cargo clippy --all-features --all-targets -- -Dwarnings` on all workspaces. So `geo-test-fixtures` won't pass the clippy guard here because `wkt` in turn depends on `geo-types`. If so, we can simply change clippy's command to `--exclude geo-test-fixtures` and that should make CI happy.

Reply:
Did that fix the build errors?

I haven't looked into it too closely, but moving jts-test-runner here (maybe even including the test files) night be a good idea.

Reply:
For me what really fixed it was to add a `wkt` entry to the `[patch.crates-io]` section of the root Cargo.toml -- it fixed it for jts and all other crates.

```
wkt = { path = "wkt" }
```

Reply:
Where does that come from? wkt is in another repository, who puts it here?

Reply:
My whole point is that we shouldn't move wkt to this repo.

Reply:
@lnicola I'm all for keeping them separate if we have a good process in place for introducing the breaking changes into the core repo without CI going bonkers :)

Reply:
@nyurik Agree that a good robust CI is what we all aim for.   To evaluate our options, could you please create a simpler PR; it's difficult to pin-point the issues / have a streamlined discussion when many different files are modified even if they're conceptually simple.  

Here's a suggestion.  Create a PR that only:

1. declares `CoordTZM<T, Z, M>`, and makes the existing `Coordinate<T>` a type-alias
2. fix any issues with using `Coordinate<T>` as a constructor / function (as it is now a type-alias)
3. make no changes to the other geometry types / algorithms
4. make no changes to any Cargo.toml / other top-level config files
5. do not bring in any other repo. in

In essence, I'm requesting the simplest step towards introducing ZM.  This already helps us to discuss:

1. is CI going to break / why?
2. what sort of vector calc. to impose on M (Add/Sub/Neg), and what are the implications?

Reply:
@rmanoka makes perfect sense, thx. I might have to take some time off to do this (getting ready for the SOTM-US conference next week), but should be able to do it soon.  In the mean time, let me know if i can help with merging the other non-breaking PRs -- they should also help with making the change simpler. Thanks for all your support!!!